zhaquirks/sonoff/swv.py

```diff
@@ -0,0 +1,49 @@
+"""Sonoff SWV - Zigbee smart water valve."""
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import EntityPlatform, EntityType, QuirkBuilder
+import zigpy.types as t
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+
+class ValveState(t.enum8):
+    """Water valve state."""
+
+    Normal = 0
+    Water_Shortage = 1
+    Water_Leakage = 2
+    Water_Shortage_And_Leakage = 3
```

Comment:
Would two separate binary sensors make more sense for this? (instead of an enum sensor)

Reply:
I would vote for 2 binary sensors. This would make it more convenient for triggers and alerts.

Reply:
https://github.com/zigpy/zha/issues/305 and https://github.com/zigpy/zha/issues/303 would be required for this.

Reply:
@TheJulianJES what about the approach in https://github.com/zigpy/zha-device-handlers/pull/3340 to have 2 binary sensors?

Reply:
@TheJulianJES While the work is being decided, can we get this without the enum so at least people can fiddle with the attributes directly using e.g. zha toolkit?

Or you reckon https://github.com/zigpy/zha/issues/305 and https://github.com/zigpy/zha/issues/303 would be handled relatively soon?

Reply:
Let's temporarily remove the quirks v2 enum sensor then. We should be able to get the quirk in the HA Core 2024.12 beta for tomorrow then.
Not sure on the timeline of the "attribute converters" yet. I hope we can get to it soon, but no promises.

Reply:
Awesome, thank you!